docs/reference/components/sources/file.cue

```diff
@@ -160,6 +161,15 @@ components: sources: file: {
 			required:    true
 			type: array: items: type: string: examples: ["\(_directory)/**/*.log"]
 		}
+		line_delimiter: {
+			common:      false
+			description: "String sequence used to separate one file line from another"
```

Comment:
Would it make more sense to be able to handle any of the standard line delimiters (`\n`, `\r\n`, and `\r`), or are there other patterns that are in use that we should also handle that can't be detected so easily?

Reply:
I think the standard line-delimiters will suffice most of the time, but Unicode does specify other characters too there:
https://en.wikipedia.org/wiki/Newline#Unicode
           
I left it as an arbitrary string so that we don't have to pre-specify all the possible characters. Also (but maybe not as important), leaving the configuration open can allow creative uses of the option -- eg: breaking down file lines by some string pattern recurring in the file.

If you prefer to make this a string enum though, I can do the change. fwiw, there are both kind of implementations out there in the wild:

https://www.elastic.co/guide/en/logstash/current/plugins-inputs-file.html#plugins-inputs-file-delimiter
https://github.com/elastic/beats/blob/v7.10.1/filebeat/_meta/config/filebeat.inputs.reference.yml.tmpl#L96-L98

---

A third option is to try and auto-detect line-endings for the file, but that might get complicated if we want to support all the possible characters, and also if the file somehow uses line endings inconsistently, it can throw off the heuristics we use. This strategy would also not work if the file starts out as empty, for example.

Reply:
I would think auto-detection, either when opening the file or just allowing any of a number of line ending patterns, would be the preferred default, making this *mostly* unnecessary. I do agree however that allowing for creative uses here would be a benefit to some users, so this is more a question of whether the default should be a fixed single pattern or something more sophisticated.

@lukesteensen, you've dealt with area this more, would appreciate your thoughts.

Reply:
My instinct was to let the line-delimiter be user-determined only (either restricted as a fixed enum or open for arbitrary entries), but I can see the appeal of an automatic handling option. Like you suggested, there's two ways we can achieve it:

1. We can auto-default to the first line-delimiter string used by the file. This is the simpler/cheaper option, and will probably cover most of the usecases.
2. We can allow all possible delimiters (either the [unicode list](https://en.wikipedia.org/wiki/Newline#Unicode) or just `\n,\r\n,\r`) and emit lines any time one of the allowed delimiters is seen. This will handle files with inconsistent line-endings too, at the added cost of needing to search all possible patterns for every line-splitting.

If users don't like this sort of "smart" handling (or they know their delimiters beforehand), they will still have the ability to choose a single line-delimiter. One thing worth noting though: if we change the default away from the current `\n` to the new auto option, it would be a breaking change and we would need to communicate the changes as such.

Also, if we are adding the auto option (either as a default or one of the choices), we need to decide the strategy to pick there from the above two. We can also expose both the variants as separate options from configuration side, though that might be one too many option.

Reply:
I am fine with leaving the auto-detection for later. If we are all in agreement about this, I can file the issue for it and we can continue the conversation there.

When we do implement it, perhaps a keyword like `auto` can stand in for that option, but we may need to rethink it in context of arbitrary strings allowed here now (or we can just treat `auto` string as special). Is there anything we should do as part of this PR, to avoid potential pain/config breakage if we are to implement the auto-detection later? There might be other configuration modes for supporting it and still enabling arbitrary strings, that I am not be seeing. Or we can drop the support for arbitrary strings and restrict the configuration to fixed string enums standing in for various standard separators, which will simplify things.

We can also choose to remove the `line_delimiter` configuration altogether from this PR and just rely on the earlier hardcoded default of `\n` for now. However, the immediate end-goal of this PR was to read non-utf8 encoded log files on Windows systems and they are mostly coupled with `\r\n` as line separator so would like to ideally handle the splitting for it too here (especially since support for wider delimiter sequence would still be there internally, for line-splitting to work consistently across encodings).